File: 백준/Gold/5052. 전화번호 목록/전화번호 목록.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());
        loop:
        for (int t = 0; t < T; t++) {
            int n = Integer.parseInt(br.readLine());
            String[] tel = new String[n];

            for (int i = 0; i < n; i++) {
                tel[i] = br.readLine();
            }

            Arrays.sort(tel);

            for (int i = 0; i < n - 1; i++) {
                String str = tel[i];
                int len = str.length();
                if (tel[i + 1].length() > len && tel[i + 1].substring(0, len).equals(str)) {
                    System.out.println("NO");
                    continue loop;
                }
            }

            System.out.println("YES");
        }
    }
}

File: Algorithem__/백준/Gold/5052. 전화번호 목록/전화번호 목록.kt
fun main() = with(System.`in`.bufferedReader()) {
    val t = readLine().toInt()
    loop@ for (tt in 0 until t) {
        val n = readLine().toInt()
        val tel = mutableListOf<String>()
        for (nn in 0 until n) {
            tel.add(readLine())
        }

        tel.sort()

        for (i in 0 until n - 1) {
            val str = tel[i]
            val target = tel[i + 1]
            if (target.length > str.length && target.substring(0, str.length) == str) {
                println("NO")
                continue@loop
            }
        }
        println("YES")
    }
}